src/haxeLanguageServer/features/haxe/DiagnosticsFeature.hx

```diff
@@ -241,6 +246,7 @@ class DiagnosticsFeature {
 	}
 
 	public function clearDiagnostics(uri:DocumentUri) {
+		removePendingRequest(uri);
```

Comment:
Hmm I wonder if we should do it here. Because we also call `clearDiagnostics` in `processDiagnosticsReply` and in this case the pending request might be a new one... Maybe we should split this method in two.

Reply:
Done